src/Compilers/CSharp/Portable/FlowAnalysis/PreciseAbstractFlowPass.cs

```diff
@@ -2366,6 +2371,55 @@ public override BoundNode VisitConditionalOperator(BoundConditionalOperator node
             return null;
         }
 
+        private BoundNode VisitRefConditionalOperator(BoundConditionalOperator node)
```

Comment:
Unfortunately that means two copies of essentially the same method. I was imagining a helper method that would only add one condition. That helper would also remove the duplication of logic at each `ref` argument.
```
BoundNode VisitConditionalOperator(BoundConditionalOperator node)
{
    ...
    if (IsConstantTrue(node.Condition))
    {
        SetState(alternativeState);
        VisitConditionalOperand(node.Alternative, node.IsByRef);
        SetState(consequenceState);
        VisitConditionalOperand(node.Consequence, node.IsByRef);
    }
    ...
}

void VisitConditionalOperand(BoundExpression expr, bool isByRef)
{
    if (isByRef)
    {
        VisitLvalue(expr);
        // exposing ref is a potential write 
        WriteArgument(expr, RefKind.Ref, method: null);
    }
    else
    {
        Visit(expr);
    }
}
```